modules/programs/vscode.nix

```diff
@@ -223,6 +230,12 @@ in {
   };
 
   config = mkIf cfg.enable {
+    programs.vscode.userDir = lib.mkDefault
```

Comment:
I put the default here, due to a default for the option will cause early evaluation, I got issues where it complained about `home.stateVersion` not being set, while it was set in a later part of the module configuration.